backends/bmv2/common/extern.h

```diff
@@ -72,95 +72,72 @@ class ExternConverter {
         type&      name;                                                        \
-        ExternConverter_##extern_name##__VA_ARGS__() :                          \
+        ExternConverter_##extern_name() :                                       \
             name(type::instance) {                                              \
             registerExternConverter(#extern_name, this); }                      \
-        static ExternConverter_##extern_name##__VA_ARGS__ singleton;            \
-        void convertExternInstance(ConversionContext* ctxt,             \
-            const IR::Declaration* c, const IR::ExternBlock* eb,        \
-            const bool& emitExternes) override;      \
+        static ExternConverter_##extern_name singleton;                         \
+        void convertExternInstance(ConversionContext* ctxt,                     \
+            const IR::Declaration* c, const IR::ExternBlock* eb,                \
+            const bool& emitExternes) override;                                 \
         Util::IJson* convertExternObject(ConversionContext* ctxt,               \
             const P4::ExternMethod* em, const IR::MethodCallExpression* mc,     \
             const IR::StatOrDecl* s, const bool& emitExterns) override; };
 
-#define EXTERN_CONVERTER_W_OBJECT_AND_INSTANCE(extern_name, ...)               \
-    class ExternConverter_##extern_name##__VA_ARGS__ : public ExternConverter { \
-        ExternConverter_##extern_name##__VA_ARGS__() {                          \
+#define EXTERN_CONVERTER_W_OBJECT_AND_INSTANCE(extern_name)                     \
+    class ExternConverter_##extern_name : public ExternConverter {              \
+        ExternConverter_##extern_name() {                                       \
             registerExternConverter(#extern_name, this); }                      \
-        static ExternConverter_##extern_name##__VA_ARGS__ singleton;            \
-        void convertExternInstance(ConversionContext* ctxt,             \
-            const IR::Declaration* c, const IR::ExternBlock* eb,        \
-            const bool& emitExterns) override;      \
+        static ExternConverter_##extern_name singleton;                         \
+        void convertExternInstance(ConversionContext* ctxt,                     \
+            const IR::Declaration* c, const IR::ExternBlock* eb,                \
+            const bool& emitExterns) override;                                  \
         Util::IJson* convertExternObject(ConversionContext* ctxt,               \
             const P4::ExternMethod* em, const IR::MethodCallExpression* mc,     \
             const IR::StatOrDecl* s, const bool& emitExterns) override; };
 
-#define EXTERN_CONVERTER_SINGLETON(extern_name, ...)                           \
-    ExternConverter_##extern_name##__VA_ARGS__                                 \
-       ExternConverter_##extern_name##__VA_ARGS__::singleton;
-
```

Comment:
I'm not sure that removing these 4 macros is a cleanup, or just makes things harder to maintain.

Reply:
The program isn't any longer...

Reply:
Actually this macro doesn't make the program shorter, the next 3 do.

Reply:
I think it is desirable to make clear which externs are objects and which are functions.  To the extent that there is anything common, one would prefer to avoid replication, as that can lead to subtle problems when there are differences -- when looking at a repeating pattern of source code, one's eyes tend to concentrate on the replicated pattern and not notice the differences, when it is the differences that are important, not the pattern.

Reply:
It may be that these 4 macros should be replaced by 2 (object and function), or 3 (if the object/instance difference is important), with the singleton declaration folded into the OBJECT or FUNCTION macro.